napari/layers/_shapes_layer/model.py

```diff
@@ -1148,7 +1233,142 @@ def to_xml_list(self, shape_type=None):
             List of xml elements defining each shape according to the
             svg specification
         """
-        return self.data.to_xml_list(shape_type=shape_type)
+        return self._data_view.to_xml_list(shape_type=shape_type)
+
+    def to_masks(self, mask_shape=None, shape_type=None):
+        """Returns N binary masks, one for each shape, embedded in an array of
+        shape `mask_shape`. Passing a `shape_type` argument leads to only mask
+        from that particular `shape_type` being returned.
+
+        Parameters
+        ----------
+        mask_shape : np.ndarray | tuple | None
+            tuple defining shape of mask to be generated. If non specified,
+            takes the max of all the vertiecs
+        shape_type : {'line', 'rectangle', 'ellipse', 'path', 'polygon'} |
+                     None, optional
+            String of shape type to be included.
+
+        Returns
+        ----------
+        masks : np.ndarray
+            Array where there is one binary mask for each shape
+        """
+        if mask_shape is None:
+            mask_shape = self.shape
+
+        if self.ndim == 2:
+            # For 2D shapes just convert current view to masks
+            masks = self._data_view.to_masks(mask_shape=mask_shape,
+                                             shape_type=shape_type)
+        elif self.broadcast:
+            # For broadcast shapes convert current view to masks and
+            # broadcast across sliced dimensions
+            slices = self._data_view.to_masks(mask_shape=mask_shape[-2:],
+                                              shape_type=shape_type)
+            masks = []
+            for m in slices:
+                masks.append(np.broadcast_to(m, mask_shape))
+            masks = np.stack(masks, axis=0)
```

Comment:
Is this functionality tested anywhere? It seems pretty brittle to me...

Reply:
this is tested in the examples, gives outputs of the right shape. What's the concern?

Reply:
probably to do with the usage of `broadcast_to` on a `slice`

also the above statement should be a list comprehension

Reply:
ah found an error here with no masks, will fix

Reply:
i've now fixed this I think